compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2657,6 +2660,20 @@ private boolean checkUnionHasSameFiniteType(LinkedHashSet<BType> memberTypes, BT
         return false;
     }
 
+    private boolean checkValueSpaceHasSameType(BFiniteType finiteType, BType baseType) {
+        if (baseType.tag == TypeTags.FINITE) {
+            return true;
```

Comment:
Please check the following. It should fail at compile-time. For whatever that was not supported, let's go with the most restrictive approach for alpha4 and relax the restriction in a separate PR.

We should make sure whatever correct comparison that worked previously should work though.